runtime/launcher.m

```diff
@@ -558,11 +574,26 @@ int main (int argc, char **argv)
 		for (int i = 1; i < argc; i++)
 			*ptr++ = argv [i];
 		*ptr = NULL;
-
-		rv = mono_main (new_argc, new_argv);
-
+        
+		if (is_extension) {
+	            rv = NSExtensionMain (new_argc, new_argv);
+	        }
+	        else {
+	            rv = mono_main (new_argc, new_argv);
+	        }
+        
 		free (new_argv);
 	}
 
 	return rv;
 }
+
+int main (int argc, char **argv)
+{
+	return xamarin_main (argc, argv, false);
+}
+
+extern "C" int extension_main (int argc, char **argv)
```

Comment:
Public functions should be prefixed with `xamarin_` (also define the function in an internal header so that you can avoid the `extern "C"` part).